app/uk/gov/hmrc/soletraderidentificationfrontend/models/SoleTraderDetails.scala

```diff
@@ -109,7 +157,31 @@ object SoleTraderDetails {
         case None => Json.obj()
       }
 
-    format.writes(soleTraderDetails) ++ formattedBusinessVerification ++ formattedIdentifiersMatch ++ formattedNinoInsights
+    val overseasTaxIdBlock: JsObject = (soleTraderDetails.optOverseasTaxIdentifier, soleTraderDetails.optOverseasTaxIdentifierCountry) match {
+      case (Some(overseasTaxId), Some(country)) => Json.obj(
+        "overseas" -> Json.obj(
+          "taxIdentifier" -> overseasTaxId,
+          "country" -> country
+        )
+      )
+      case (None, None) => Json.obj()
+      case _ => throw new InternalServerException("Error: Invalid combination of tax identifier and country")
+    }
+
+
+    format.writes(soleTraderDetails) ++ formattedBusinessVerification ++ formattedIdentifiersMatch ++ formattedNinoInsights ++ overseasTaxIdBlock
   }
 
+  private def determineOverseasTaxIdentifierDetails(optOverseasTaxI: Option[String],
```

Comment:
It is probably not going to be around for long, but I'd prefer "optOverseasTaxId" to "optOverseasTaxI".

Reply:
done